### PR TITLE
Make more progress on IDE0044 ("Make field readonly")

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs [hunk 1]
```diff
@@ -16,18 +16,18 @@ namespace System.Reflection
     internal sealed partial class RuntimeConstructorInfo : ConstructorInfo, IRuntimeMethodInfo
     {
         #region Private Data Members
-        private volatile RuntimeType m_declaringType;
```
Comment (stephentoub) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs:19]: @jkotas, I couldn't see any reason why this should be volatile.  Is there one?  If not, I'll leave my change to make it readonly.  If there is, I'll revert and comment as to why.

Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs:19]: I agree that there is no reason for `volatile` here.

I have looked into history to find out why this volatile was added. It was added based on ConcurrencyCop static analysis tool. The change added hundreds volatile keywords all over the place. I guess that this one was added by mistake.

Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs:19]: Thanks. Can you point me to the changeset? Might be worth a quick audit to see if others might have been added erroneously.

Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs:19]: https://vstfdevdiv/DevDiv2/DevDiv/_versionControl/changeset/16007/ and https://vstfdevdiv/DevDiv2/DevDiv/_versionControl/changeset/19093/

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilderInstantiation.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ internal static MethodInfo MakeGenericMethod(MethodInfo method, Type[] inst)
 
         #region Private Data Members
         internal MethodInfo m_method;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilderInstantiation.cs:23]:
```suggestion
        internal readonly MethodInfo m_method;
```
?

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeEventBuilder.cs [hunk 3]
```diff
@@ -99,10 +99,10 @@ protected override void SetCustomAttributeCore(CustomAttributeBuilder customBuil
         }
 
         // These are package private so that TypeBuilder can access them.
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeEventBuilder.cs:101]:
```suggestion
```

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/XXXOnTypeBuilderInstantiation.cs [hunk 4]
```diff
@@ -18,7 +18,7 @@ internal static MethodInfo GetMethod(MethodInfo method, TypeBuilderInstantiation
 
         #region Private Data Members
         internal MethodInfo m_method;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/XXXOnTypeBuilderInstantiation.cs:20]:
```suggestion
        internal readonly MethodInfo m_method;
```

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/XXXOnTypeBuilderInstantiation.cs [hunk 5]
```diff
@@ -94,7 +94,7 @@ internal static ConstructorInfo GetConstructor(ConstructorInfo Constructor, Type
 
         #region Private Data Members
         internal ConstructorInfo m_ctor;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/XXXOnTypeBuilderInstantiation.cs:96]:
```suggestion
        internal readonly ConstructorInfo m_ctor;
```

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs [hunk 6]
```diff
@@ -37,8 +37,8 @@ internal static Type MakeGenericType(Type type, Type[] typeArguments)
         #endregion
 
         #region Private Data Members
-        private Type m_type;
-        private Type[] m_inst;
+        private readonly Type m_type;
+        private readonly Type[] m_inst;
         private string? m_strFullQualName;
         internal Hashtable m_hashtable;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs:43]:
```suggestion
        internal readonly Hashtable m_hashtable;
```

File: src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs [hunk 7]
```diff
@@ -9,7 +9,7 @@ namespace System.Reflection
     internal abstract class RuntimeFieldInfo : FieldInfo
     {
         #region Private Data Members
-        private BindingFlags m_bindingFlags;
+        private readonly BindingFlags m_bindingFlags;
         protected RuntimeTypeCache m_reflectedTypeCache;
         protected RuntimeType m_declaringType;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs:14]:
```suggestion
        protected readonly RuntimeTypeCache m_reflectedTypeCache;
        protected readonly RuntimeType m_declaringType;
```

**Review by jkotas — APPROVED** [review]

**Review by marek-safar — APPROVED** [review]

**Pull request merged**